File: Arrays & Maths/majorityElement.py
class Solution:
    def majorityElement(self, nums):
        ele = nums[0]
        freq = 1

        for i in range(1, len(nums)):
            if freq == 0:
                ele = nums[i]
                freq = 1
            elif ele != nums[i]:
                freq -= 1
            else:
                freq += 1

        return ele

File: Arrays & Maths/MajorityElement.java
public class MajorityElement {
    int majorityElement(int[] nums) {
        int ele = nums[0];
        int freq = 1;

        for (int i = 1; i < nums.length; i++) {
            if (freq == 0) {
                ele = nums[i];
                freq = 1;
            } else if (ele != nums[i]) {
                freq--;
            } else {
                freq++;
            }
        }
        return ele;
    }

    public static void main(String[] args) {

    }
}
